core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -269,28 +269,28 @@ class KafkaServer(val config: KafkaConfig, time: Time = SystemTime, threadNamePr
   }
 
   private def initZk(): ZkUtils = {
-    info("Connecting to zookeeper on " + config.zkConnect)
+    info(s"Connecting to zookeeper on ${config.zkConnect}")
 
-    val chroot = {
-      if (config.zkConnect.indexOf("/") > 0)
-        config.zkConnect.substring(config.zkConnect.indexOf("/"))
-      else
-        ""
+    val chrootIndex = config.zkConnect.indexOf("/")
+    val chrootOption = {
+      if (chrootIndex > 0) Some(config.zkConnect.substring(chrootIndex))
+      else None
     }
 
-    val secureAclsEnabled = JaasUtils.isZkSecurityEnabled() && config.zkEnableSecureAcls
+    val secureAclsEnabled = config.zkEnableSecureAcls
```

Comment:
You have good reason to be confused. The existing code has a redundant clause, but it's actually correct as well. My bad.